src/swarm-engine/Swarm/Game/Robot/Concrete.hs

```diff
@@ -60,6 +66,7 @@ instance ToSample Robot where
       mkRobot
```

Comment:
It seems like this is the only place we use `mkRobot` to create a `Robot`, as opposed to a `TRobot`.  I wonder if we should just specialize `mkRobot` to only produce `TRobot`, which would cut down on all the annoying `()` arguments we have to pass to it right now.  Then here, we could create `sampleBase` by composing `mkRobot` and `instantiateRobot`?  Would that work?

Reply:
Good idea, that works.

I wonder if at this point it would make sense to consolidate the three fields that are `()` for `TRobot` into their own record that itself would be `()` for `TRobot`, since no `TRobot` ever needs to access those fields individually.